### PR TITLE
Add Django 4 testing and remove unsupported Python and Django versions

### PR DESCRIPTION
**Pull request by vchrisb**
* removed unsupported Django Versions `2.2, 3.0, 3.1` and Python Version `3.6`
* added Django `4.0` and Python `3.10`
* Updated Mysql and Postgresql Versions
* Updated Github Actions
* fixed pre-commit.ci by adding basic pre-commit-config

### CODE REVIEW

File: .github/workflows/test.yml
Comment (clintonb) [inline .github/workflows/test.yml:13]: “Main” is missing.

File: tox.ini
Comment (clintonb) [inline tox.ini:20]: djmain

**Review by clintonb — CHANGES_REQUESTED** [review]
Fix the test config, and this should be all set.

**Review by clintonb — APPROVED** [review]

**Comment by vchrisb** [comment]
please don't merge yet. There is still a mistake...

File: tox.ini
Comment (vchrisb) [inline tox.ini:18]: This should be django32

**Review by clintonb — CHANGES_REQUESTED** [review]
Awaiting change from engineer.

**Comment by clintonb** [comment]
@vchrisb can you update the classifiers in setup.py, too?

**Review by clintonb — APPROVED** [review]

**Comment by vchrisb** [comment]
I'll create another pr to increase the test coverage once this is merged.

**Pull request merged**
